WasatchNET/EEPROM.cs

```diff
@@ -1641,61 +1653,71 @@ public EEPROMJSON toJSON()
+                json.ROIVertRegionEnds[1] = ROIVertRegionEnd[1];
+                json.ROIVertRegionEnds[2] = ROIVertRegionEnd[2];
+            }
             json.LaserPowerCoeffs = new double[4];
-            json.LaserPowerCoeffs[0] = laserPowerCoeffs[0];
-            json.LaserPowerCoeffs[1] = laserPowerCoeffs[1];
-            json.LaserPowerCoeffs[2] = laserPowerCoeffs[2];
-            json.LaserPowerCoeffs[3] = laserPowerCoeffs[3];
+            if (laserPowerCoeffs != null)
+            {
+                json.LaserPowerCoeffs[0] = laserPowerCoeffs[0];
+                json.LaserPowerCoeffs[1] = laserPowerCoeffs[1];
+                json.LaserPowerCoeffs[2] = laserPowerCoeffs[2];
+                json.LaserPowerCoeffs[3] = laserPowerCoeffs[3];
+            }
             json.MaxLaserPowerMW = maxLaserPowerMW;
             json.MinLaserPowerMW = minLaserPowerMW;
             json.ExcitationWavelengthNM = laserExcitationWavelengthNMFloat;
             json.AvgResolution = avgResolution;
             json.BadPixels = new int[15];
-            json.BadPixels[0] = badPixels[0];
-            json.BadPixels[1] = badPixels[1];
-            json.BadPixels[2] = badPixels[2];
-            json.BadPixels[3] = badPixels[3];
-            json.BadPixels[4] = badPixels[4];
-            json.BadPixels[5] = badPixels[5];
-            json.BadPixels[6] = badPixels[6];
-            json.BadPixels[7] = badPixels[7];
-            json.BadPixels[8] = badPixels[8];
-            json.BadPixels[9] = badPixels[9];
-            json.BadPixels[10] = badPixels[10];
-            json.BadPixels[11] = badPixels[11];
-            json.BadPixels[12] = badPixels[12];
-            json.BadPixels[13] = badPixels[13];
-            json.BadPixels[14] = badPixels[14];
+            if (badPixels != null)
+            {
+                json.BadPixels[0] = badPixels[0];
+                json.BadPixels[1] = badPixels[1];
+                json.BadPixels[2] = badPixels[2];
+                json.BadPixels[3] = badPixels[3];
+                json.BadPixels[4] = badPixels[4];
+                json.BadPixels[5] = badPixels[5];
+                json.BadPixels[6] = badPixels[6];
+                json.BadPixels[7] = badPixels[7];
+                json.BadPixels[8] = badPixels[8];
+                json.BadPixels[9] = badPixels[9];
+                json.BadPixels[10] = badPixels[10];
+                json.BadPixels[11] = badPixels[11];
+                json.BadPixels[12] = badPixels[12];
+                json.BadPixels[13] = badPixels[13];
+                json.BadPixels[14] = badPixels[14];
+            }
             json.UserText = userText;
             json.ProductConfig = productConfiguration;
             json.RelIntCorrOrder = intensityCorrectionOrder;
-            if (json.RelIntCorrOrder > 0)
+            if (json.RelIntCorrOrder > 0 && intensityCorrectionCoeffs != null)
             {
                 json.RelIntCorrCoeffs = new double[intensityCorrectionCoeffs.Length];

```

Comment:
I suspect this may be the source of the customer's exception.  Previously it was possible (at least within this function) for intensityCorrectionCoeffs to be null at this line, at which point we'd be calling the Length attribute on a null object.